From 1145ba0cd28185ee953830c43bdfb2a3aff41253 Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <bbhushan2@marvell.com>
Date: Tue, 12 May 2020 16:45:53 +0530
Subject: [PATCH 0500/1921] octeontx2: gti: Fix task stack pointer corruption

GTI NMI exception handling was corrupting task stack
pointer (tsk <--> x28). As only x29 is used, so this
patch now restores only x29 and not corrupting x28.

Change-Id: Iba7c4e810448a107d8eb97b3e94f787826566419
Signed-off-by: Bharat Bhushan <bbhushan2@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/28388
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/kernel/entry.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/kernel/entry.S b/arch/arm64/kernel/entry.S
index 354a4bd380f7..4054fa222c82 100644
--- a/arch/arm64/kernel/entry.S
+++ b/arch/arm64/kernel/entry.S
@@ -204,7 +204,7 @@ alternative_cb_end
 	.if	\el == 0
 	mov	x29, xzr
 	.else
-	ldp	x28, x29, [sp, #S_X28]
+	ldr	x29, [sp, #S_X28+8]
 	.endif
 	.endif	/* \exc_el3 == 0 */
 #else
-- 
2.31.1

